### PR TITLE
initial work, with test and CI

### PR DESCRIPTION
**Pull request by chentex**
Signed-off-by: Vicente Zepeda Mas <vzepedamas@suse.com>

- Main workflow added. Usage `caaasp-init -c /etc/kubic/kubic-init.yaml`.

- CI steps

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -18,26 +18,94 @@ import (
...
-	"github.com/spf13/viper"
 )
 
-var cfgFile string
+var (
+	cfgFile              string
+	registryConfigFolder = "/etc/docker"
+)
+
+//   {{ $a := substract 2 1}}
```
Comment (bergmannf) [inline cmd/root.go]: Seems to be a remnant of some previous version.

**Review by bergmannf — APPROVED** [review]
LGTM, but I am probably not a good judge for the `go` code concerning good style / idiomatic-ness, as I haven't written much code in `go` yet.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+		git \
+		go \
+		golang-github-cpuguy83-go-md2man \
+		make \
+		tar
+
+ENV GOPATH /go
+ENV PATH $GOPATH/bin:$PATH
+RUN go get -u golang.org/x/lint/golint && \
+	go get -u github.com/vbatts/git-validation && type git-validation
```
Comment (mssola) [inline Dockerfile:16]: Even if it may get a bit messy, I'd merge all `RUN`s into a single statement if possible.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+	github-release release -u kubic-project -r caasp-init --tag $(VERSION)  --name $(VERSION) -s $(GITHUB_TOKEN) -d "$(CHANGE)"
+	github-release upload -u kubic-project -r caasp-init --tag $(VERSION)  --name caasp-init-linux.tgz --file release/caasp-init-linux.tgz -s $(GITHUB_TOKEN)
+	github-release upload -u kubic-project -r caasp-init --tag $(VERSION)  --name caasp-init-macos.tgz --file release/caasp-init-macos.tgz -s $(GITHUB_TOKEN)
+	github-release upload -u kubic-project -r caasp-init --tag $(VERSION)  --name caasp-init-windows.tgz --file release/caasp-init-windows.tgz -s $(GITHUB_TOKEN)
+
+MANPAGES_MD := $(wildcard doc/man/*.md)
+MANPAGES    := $(MANPAGES_MD:%.md=%)
+
+doc/man/%.1: doc/man/%.1.md
+	$(GO_MD2MAN) -in $< -out $@
```
Comment (mssola) [inline Makefile]: I'd recommend taking a look at what we are doing for man pages on [portusctl](https://github.com/openSUSE/portusctl/blob/master/Makefile#L120). That is, md2man usually outputs really awkward man pages (with lots of spaces and so on). As you can see, for that project, I hacked a quick go script that filters unneeded clutter.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
+# caasp-init
+
+[![Release](https://img.shields.io/github/release/kubic-project/caasp-init.svg)](https://github.com/kubic-project/caasp-init/releases/latest)
+[![Build Status](https://img.shields.io/travis/kubic-project/caasp-init/master.svg)](https://travis-ci.org/kubic-project/caasp-init)
+[![codecov](https://codecov.io/gh/kubic-project/caasp-init/branch/master/graph/badge.svg)](https://codecov.io/gh/kubic-project/caasp-init)
+
+![License: Apache 2.0](https://img.shields.io/github/license/kubic-project/caasp-init.svg)
+[![Go Report Card](https://goreportcard.com/badge/github.com/kubic-project/caasp-init)](https://goreportcard.com/report/github.com/kubic-project/caasp-init)
+
+Initialize docker daemon mirrors configuration
```
Comment (mssola) [inline README.md]: Period at the end :wink:

File: devel.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Development environment for `caasp-init`
```
Comment (mssola) [inline devel.md]: Some of this stuff is redundant with some info from the README.md file. Plus, I don't think is worth giving this information in a separate file, I'd merge this info with the README.md file directly.

**Review by mssola — COMMENTED** [review]
Mostly LGTM. Great job :clap:

File: Makefile (hunk 3 above)
Comment (chentex) [reply Makefile]: Nice hack!

**Review by mssola — APPROVED** [review]
LGTM

**Pull request merged**